catalog/ui/catalog-ui-search/src/main/webapp/js/Common.js

```diff
@@ -20,6 +20,8 @@ define([
     var format = 'DD MMM YYYY HH:mm:ss.SSS';
 
     return {
+        //randomly generated guid guaranteed to be unique ;)
+        undefined: '2686dcb5-7578-4957-974d-aaa9289cd2f0',
```

Comment:
But really, this is used to handle the need for a unique key in a map that can technically have any value as a key and also needs to be able to support blank as well as undefined (as two distinct keys).